kernel/bus/Legacy.cpp

```diff
@@ -12,6 +12,26 @@ Iteration legacy_scan(IterationCallback<LegacyAddress> callback)
         return Iteration::STOP;
     }
 
+    if (callback(LEGACY_ATA0) == Iteration::STOP)
+    {
+        return Iteration::STOP;
+    }
+
+    if (callback(LEGACY_ATA1) == Iteration::STOP)
+    {
+        return Iteration::STOP;
+    }
+
+    if (callback(LEGACY_ATA2) == Iteration::STOP)
+    {
+        return Iteration::STOP;
+    }
+
+    if (callback(LEGACY_ATA3) == Iteration::STOP)
+    {
+        return Iteration::STOP;
+    }
```

Comment:
Maybe we can have a static method in LegacyATA to check if they exist like:

```c++
if (LegacyATA::exist(LEGACY_ATA2) && callback(LEGACY_ATA2) == Iteration::STOP)
{
    return Iteration::STOP;
}
```

Reply:
Good idea, will do that. Will call the IDENTIFY command twice then for each command though, since i still need the information there